frontend/src/app/modules/user-profile/user-profile-page/user-profile.component.ts

```diff
@@ -0,0 +1,219 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { BaseComponent } from '@core/base/base.component';
+import { AuthService } from '@core/services/auth.service';
+import { NotificationService } from '@core/services/notification.service';
+import { SpinnerService } from '@core/services/spinner.service';
+import { UserService } from '@core/services/user.service';
+import { faEye, faEyeSlash, faPen } from '@fortawesome/free-solid-svg-icons';
+import { ValidationsFn } from '@shared/helpers/validations-fn';
+import { finalize, takeUntil } from 'rxjs';
+
+import { UpdateUserNamesDto } from 'src/app/models/user/update-userNames.dto';
+import { UpdateUserNotificationsDto } from 'src/app/models/user/update-userNotifications.dto';
+import { UpdateUserPasswordDto } from 'src/app/models/user/update-userPassword.dto';
+import { UserDto } from 'src/app/models/user/user-dto';
+
+@Component({
+    selector: 'app-user-profile',
+    templateUrl: './user-profile.component.html',
+    styleUrls: ['./user-profile.component.sass'],
+})
+export class UserProfileComponent extends BaseComponent implements OnInit, OnDestroy {
+    public squirrelNotification: boolean;
+
+    public emailNotification: boolean;
+
+    public openEyeIcon = faEye;
+
+    public penIcon = faPen;
+
+    public closeEyeIcon = faEyeSlash;
+
+    public currentUser: UserDto;
+
+    public userNamesForm: FormGroup = new FormGroup({});
+
+    public passwordForm: FormGroup = new FormGroup({});
+
+    public notificationsForm: FormGroup = new FormGroup({});
+
+    constructor(
+        private fb: FormBuilder,
+        private userService: UserService,
+        private notificationService: NotificationService,
+        private authService: AuthService,
+        private spinner: SpinnerService,
+    ) {
+        super();
+    }
+
+    public ngOnInit() {
+        this.fetchCurrentUser();
+    }
+
+    private fetchCurrentUser() {
+        this.spinner.show();
+
+        this.authService
+            .getUser()
+            .pipe(
+                takeUntil(this.unsubscribe$),
+                finalize(() => this.spinner.hide()),
+            )
+            .subscribe(
+                (user) => {
+                    this.currentUser = user as UserDto;
+                    this.initializeForms();
+                },
+                () => {
+                    this.notificationService.error('Failed to fetch current user');
+                },
+            );
+    }
+
+    public override ngOnDestroy() {
+        super.ngOnDestroy();
+    }
```

Comment:
no need to override if you just call `super`

Reply:
removed